ios/SubscriptionsDB.swift

```diff
@@ -49,48 +55,72 @@ class SubscriptionsDB {
   func add(channelName: String) -> Bool {
     guard !exists(channelName) else { return false }
     
-    let insert = table.insert(
-      columnChannelName <- channelName,
-      columnNotificationsEnabled <- true)
-  
-    do { try db.run(insert) }
-    catch { return false }
+    do {
+      try db.transaction(block: {
+        let insert = self.table.insert(
+          self.columnChannelName <- channelName,
+          self.columnNotificationsEnabled <- true)
+        
+        try self.db.run(insert)
```

Comment:
I'm wrapping all the operations into transactions, it's a standard way of working with dbs. If for any reason the code between the transaction fails, the DB will rollback any change made to the db and we wont have an inconsistent state.